src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -310,11 +316,18 @@ public void initialize() {
     autoChannelListView.setSelectionModel(new NoSelectionModel<>());
     autoChannelListView.setFocusTraversable(false);
     autoChannelListView.setItems(preferencesService.getPreferences().getChat().getAutoJoinChannels());
-    autoChannelListView.setCellFactory(param -> uiService.<RemovableListCellController>loadFxml("theme/settings/removable_cell.fxml"));
+    autoChannelListView.setCellFactory(param -> uiService.<RemovableListCellController<String>>loadFxml("theme/settings/removable_cell.fxml"));
     autoChannelListView.getItems().addListener((ListChangeListener<String>) c -> preferencesService.storeInBackground());
-    autoChannelListView.managedProperty().bind(autoChannelListView.visibleProperty());
+    JavaFxUtil.bindManagedToVisible(autoChannelListView);
     autoChannelListView.visibleProperty().bind(Bindings.createBooleanBinding(() -> !autoChannelListView.getItems().isEmpty(), autoChannelListView.getItems()));
 
+    mirrorURLsListView.setSelectionModel(new NoSelectionModel<>());
+    mirrorURLsListView.setFocusTraversable(false);
+    mirrorURLsListView.setItems(preferencesService.getPreferences().getMirror().getMirrorURLs());
+    mirrorURLsListView.setCellFactory(param -> uiService.<RemovableListCellController<URI>>loadFxml("theme/settings/removable_cell.fxml"));
+    mirrorURLsListView.getItems().addListener((ListChangeListener<URI>) c -> preferencesService.storeInBackground());
+    mirrorURLsListView.getItems().addListener((ListChangeListener<URI>) c -> mirrorURLsListView.setVisible(!mirrorURLsListView.getItems().isEmpty()));
```

Comment:
rather than two listeners it is probably better to just make both into a function and then call that function in a single listener.

Also will want to use JavaFxUtil.addListener for thread safety just in case.

Reply:
I switched it to use JavaFxUtil, but I didn't see anywhere in the code that used a method rather than a lambda. Lambda is fine IMO. It even makes it easier to read because the code for the listener is right there where the listener is added.